localstack/utils/docker.py

```diff
@@ -163,85 +263,131 @@ def copy_into_container(
         LOG.debug("Copying into container with cmd: %s", cmd)
         safe_run(cmd)
 
-    def get_image_entrypoint(self, docker_image: str) -> str:
-        """Get the entry point for the given image"""
-        LOG.debug("Getting the entrypoint for image: %s", docker_image)
+    def copy_from_container(
+        self, container_name: str, local_path: str, container_path: str
+    ) -> None:
         cmd = self._docker_cmd()
-        cmd += [
-            "image",
-            "inspect",
-            '--format="{{ .Config.Entrypoint }}"',
-            docker_image,
-        ]
+        cmd += ["cp", f"{container_name}:{container_path}", local_path]
+        LOG.debug("Copying from container with cmd: %s", cmd)
+        safe_run(cmd)
+
+    def pull_image(self, docker_image: str) -> None:
+        """Pulls a image with a given name from a docker registry"""
+        cmd = self._docker_cmd()
+        cmd += ["pull", docker_image]
+        LOG.debug("Pulling image with cmd: %s", cmd)
+        try:
+            safe_run(cmd)
+        except subprocess.CalledProcessError as e:
+            raise ContainerException(
+                "Docker process returned with errorcode %s" % e.returncode, e.stdout, e.stderr
+            )
 
+    def get_docker_image_names(self, strip_latest=True, include_tags=True):
+        format_string = "{{.Repository}}:{{.Tag}}" if include_tags else "{{.Repository}}"
+        cmd = self._docker_cmd()
+        cmd += ["images", "--format", format_string]
         try:
-            run_result = safe_run(cmd)
+            output = safe_run(cmd)
+
+            image_names = output.splitlines()
+            if strip_latest:
+                suffix = ":latest"
+                for image in list(image_names):
+                    if image.endswith(suffix):
+                        image_names.append(image[: -len(suffix)])
+            return image_names
+        except Exception as e:
+            LOG.info('Unable to list Docker images via "%s": %s' % (cmd, e))
+            return []
+
+    def get_container_logs(self, container_name_or_id: str, safe=False) -> str:
+        """Get all logs of a given container"""
+        cmd = self._docker_cmd()
+        cmd += ["logs", container_name_or_id]
+        try:
+            return safe_run(cmd)
         except subprocess.CalledProcessError as e:
-            if "No such image" in e.stdout.decode(config.DEFAULT_ENCODING):
-                raise NoSuchImage(docker_image, stdout=e.stdout, stderr=e.stderr)
+            if not safe:
```

Comment:
I see that you've copied the code from above, but wondering if we had this wrong to begin with. I guess we should change this to:
```
if safe:
    return ""
```
?